### ISSUE TITLE
Workspace support?

### ISSUE THREAD

**Issue by ErichDonGubler** [Bunogi/cargo-makedocs#6]
Inlining the `repro.sh` I have in my [MRE repo](https://gitlab.com/erichdongubler-mre/2019-06-cargo-makedocs-workspace) for this issue:

---

`cargo-makedocs` looks like an interesting tool, but it doesn't seem to work with Cargo workspaces yet.

Steps to reproduce:

1. Create a new workspace with a single member -- for the sake of example, let's call this `binary` and make it a new Cargo binary.

```sh
dir="$(CDPATH= cd -- "$(dirname -- "$0")" && pwd -P)"
cd "$dir"
NEW_WORKSPACE_NAME="new-workspace"
NEW_MEMBER_NAME="binary"
if [ -e "$NEW_WORKSPACE_NAME" ]; then
	rm -rf "$NEW_WORKSPACE_NAME" || exit $?
fi
cargo new "$NEW_WORKSPACE_NAME" || exit $?
cd "$NEW_WORKSPACE_NAME" || exit $?
rm -rf *
cat > Cargo.toml <<-EOF || exit $?
	[workspace]
	members = ['$NEW_MEMBER_NAME']
EOF
cargo new --bin $NEW_MEMBER_NAME || exit $?
```

2. Build the workspace. This shouldn't fail.

```sh
cargo build || exit $?
```

3. Sanity check: run `cargo makedocs` on workspace root, expecting it to fail because a concrete workspace member should be used to generate docs.

```sh
TEMP_CARGO_INSTALL_ROOT="$dir/cargo-install-root"
if [ -e "$TEMP_CARGO_INSTALL_ROOT" ]; then
	rm -rf "$TEMP_CARGO_INSTALL_ROOT" || exit $?
fi
mkdir "$TEMP_CARGO_INSTALL_ROOT" || exit $?
export PATH="$PATH:$TEMP_CARGO_INSTALL_ROOT/bin"
cargo install --root "$TEMP_CARGO_INSTALL_ROOT" cargo-makedocs || exit $?
cargo makedocs && { echo "expected \`cargo makedocs\` to fail at workspace root!"; exit 1; }
echo "^ This failure is expected -- it's the workspace root. :)"
```

4. Change directory into the newly built `binary` workspace member. Let's try to use `makedocs` on it, expecting this to work.

```sh
cd "$NEW_MEMBER_NAME"
cargo makedocs || exit $?
```

Unfortunately, the above invocation fails, complaining that `Cargo.lock` couldn't be found. `Cargo.lock` DOES exist, but at the workspace root and not this working directory. This seems to indicate that `makedocs` simply isn't aware of workspaces yet.

**Comment by Bunogi** [comment]
You're right, workspaces isn't a feature I've really used much and I partially forgot it existed, heh.

Looking at the behaviour of running `cargo doc` in the plain workspace, it will document all dependencies for all crates, and running it in a crate will cause it to only document that crate's dependencies. `cargo-makedocs` should do the same. I'll get to work on it, shouldn't take too long.

**Comment by Bunogi** [comment]
I've rolled up a fix in 57dd5f0, which seems to work. It passes the repro and as far as I can tell works just fine. If you could confirm that it works in your original workspace I'll happily publish it.

**Comment by ErichDonGubler** [comment]
That particular commit works for me when I do `cargo install -f --git https://github.com/Bunogi/cargo-makedocs --rev 57dd5f0`, so I think this can be resolved! :) Thanks for the fast turnaround.
